src/devtools/client/debugger/src/actions/project-text-search.js

```diff
@@ -80,13 +80,18 @@ export function searchSources(cx, query) {
     const validSources = getSourceList(getState()).filter(
       source => !ThreadFront.isMinifiedSource(source.id) && !isThirdParty(source)
     );
-    for (const source of validSources) {
-      if (cancelled) {
-        return;
-      }
-      await dispatch(loadSourceText({ cx, source }));
-      await dispatch(searchSource(cx, source.id, query));
-    }
+    await Promise.all(
+      validSources.map(async source => {
+        if (cancelled) {
+          return;
+        }
+        await dispatch(loadSourceText({ cx, source }));
+        if (cancelled) {
+          return;
+        }
+        await dispatch(searchSource(cx, source.id, query));
+      })
+    );
```

Comment:
I imagine this could be quite a large number of files on any arbitrary project. In practice, maybe it makes sense to request in batches of 100 or 1000 rather than EVERYTHING at once.

Reply:
yeah - chunking is definitely advisable here